FerramAerospaceResearch/LEGACYferram4/FARControllableSurface.cs

```diff
@@ -545,6 +545,7 @@ public void CalculateSurfaceFunctions()
 
             Vector3 CoM = Vector3.zero;
             float mass = 0;
+            UpdateShipPartsList();
```

Comment:
`FARWingAerodynamicModel` also uses the part list. This should be in
https://github.com/dkavolis/Ferram-Aerospace-Research/blob/95e127ae140b4be9699da8783d24dd8db726d753/FerramAerospaceResearch/LEGACYferram4/FARPartModule.cs#L69-L72
before `OnVesselPartsChange` is invoked and
https://github.com/dkavolis/Ferram-Aerospace-Research/blob/95e127ae140b4be9699da8783d24dd8db726d753/FerramAerospaceResearch/LEGACYferram4/FARPartModule.cs#L81
can be removed

Reply:
I've run some more tests to see exactly when the `null` parts are showing up in relation to the `OnVesselPartsChange` events and it turns out that they're actually occurring from the
```
if (justStarted)
    CalculateSurfaceFunctions();
```
call on lines 469 - 470 of FARControllableSurface.cs instead, which can occur just after a plane hits the terrain and the section of the plane containing the part becomes a new vessel.
Adding a `UpdateShipPartsList` to this `if` block seems to fix the issue.

(`UpdateShipPartsList` should still be called before `OnVesselPartsChange` is invoked though to ensure that the list is up-to-date for `CalculateSurfaceFunctions` and `UpdateThisWingInteractions`.)